src/Serializer/AbstractItemNormalizer.php

```diff
@@ -142,8 +142,13 @@ protected function getAllowedAttributes($classOrObject, array $context, $attribu
 
             if (
```

Comment:
```php
if (
    $this->isAllowedAttribute($classOrObject, $propertyName, null, $context) &&
    (
        isset($context['api_normalize']) && $propertyMetadata->isReadable() ||
        isset($context['api_denormalize']) && ($propertyMetadata->isWritable() || !is_object($classOrObject) && $propertyMetadata->isInitializable())
    )
) {
    $allowedAttributes[] = $propertyName;
}
```

Reply:
Long lines should be avoided, isn't it?

Reply:
Split across more lines != easier to read :smile:

Reply:
I think many nested levels of brackets, especially with uneven indentation levels, is much worse than long lines.

Reply:
✅